python/pyarrow/tests/test_memory.py

```diff
@@ -133,3 +133,24 @@ def test_env_var():
     if should_have_mimalloc:
         check_env_var("mimalloc", ["mimalloc"])
     check_env_var("nonexistent", possible_backends, expect_warning=True)
+
+
+def test_specific_memory_pools():
```

Comment:
Will this case fail if we build a wheel without the specific optional allocator enabled (based on the should have flag [definitions](https://github.com/apache/arrow/blob/053f4742c7fde9c1e3b3a0858c4de6dc05efec98/python/pyarrow/tests/test_memory.py#L29-L30))?

If yes we may want to find a better way to retrieve the built allocator in a follow-up.

Reply:
Unless the case happens, I don't think we should worry.